packages/react-components/react-teaching-popover-preview/src/components/TeachingPopoverCarouselNav/useTeachingPopoverCarouselNav.tsx

```diff
@@ -21,6 +22,31 @@ export const useTeachingPopoverCarouselNav_unstable = (
   const focusableGroupAttr = useFocusableGroup({ tabBehavior: 'limited' });
   const totalPages = useTeachingPopoverCarouselContext_unstable(context => context.totalPages);
   const currentPage = useTeachingPopoverCarouselContext_unstable(context => context.currentPage);
+  const setCurrentPage = useTeachingPopoverCarouselContext_unstable(context => context.setCurrentPage);
+
+  let handleKeyDown = useEventCallback((ev: React.KeyboardEvent<HTMLDivElement>) => {
+    props.onKeyDown?.(ev);
+
+    if (ev.isDefaultPrevented()) {
+      return;
+    }
+
+    const key = ev.key;
+
+    if (key === ArrowUp || key === ArrowLeft) {
+      let prevPage = Math.max(0, currentPage - 1);
+      setCurrentPage(prevPage);
+    }
+
+    if (key === ArrowDown || key === ArrowRight) {
+      let nextPage = Math.min(totalPages - 1, currentPage + 1);
+      setCurrentPage(nextPage);
+    }
+  });
+
+  if (props.onKeyDown) {
```

Comment:
`mergeCallbacks()` [handles the null case](https://github.com/microsoft/fluentui/blob/master/packages/react-components/react-utilities/src/utils/mergeCallbacks.ts#L24) so you don't need this check.

```suggestion

```